src/comparator/resource_database.py

```diff
@@ -25,45 +24,50 @@ def __init__(self):
         self.types = {}
         self.patterns = {}
 
-    def register_resource(self, resource_message: ResourceDescriptor):
+    def register_resource(self, resource_with_location):
         """ Register a resource in the database. """
```

Comment:
The type of `resource_with_location` is `Resource` from `wrappers` file. While direct import will result in circular dependencies, so we removed the typing here.